src/java.base/share/classes/jdk/internal/util/random/RandomSupport.java

```diff
@@ -1210,15 +1210,16 @@ public static double computeNextExponentialSoftCapped(RandomGenerator rng, doubl
             return DoubleZigguratTables.exponentialX[(int)i] * (U1 >>> 1);
         }
         // We didn't use the upper part of U1 after all.  We'll probably be able to use it later.
-        if (maxValue <= 0.0) {
-            return 0.0;
+        if (maxValue <= DoubleZigguratTables.exponentialX0) {
+            return DoubleZigguratTables.exponentialX0;
```

Comment:
Chris, could you explain why this change to this if block is necessary? Guy notes that (I've paraphrased):

> I can see that (without this proposed change here), if `maxValue` is not greater than `DoubleZigguratTables.exponentialX0`, then the subsequent computation for `maxExtraMinus1`, a few lines below using `Math.round()`, will compute 1 or 2 for the value of `maxExtraMinus1`. But with this proposed change, it just returns `DoubleZigguratTables.exponentialX0`, which by the contract of `computeNextExponentialSoftCapped` is okay (the doc says "{@code maxValue} is a "soft" cap in that a value larger than {@code maxValue} may be returned in order to save a calculation" and I remember that that is true). So that is also okay. The motivation for doing so would be that it saves computation time overall. The part I don't quite understand yet is the judgment that it will in fact save computation time overall. If you take advantage of the "soft cap" then it could cause additional iteration of the loop in `computeNextGaussian` where this `computeNextExponentialSoftCapped` method gets called.

Reply:
Updated to return `maxValue` instead.